### ISSUE TITLE
Platform Translation

### ISSUE THREAD

**Issue by julienM77** [linkwarden/linkwarden#216]
I didn't see any mention in the rodmap of the site's multilingual capability (EN, FR, DE, etc).

Do you plan to implement this in the future?

**Comment by DonMercedesMoped** [comment]
Would be interested to contribute (DE <=> EN).

**Comment by kgnfth** [comment]
Would be interested to contribute (NL <=> EN) & (TR <=> EN).

**Comment by daniel31x13** [comment]
Gonna add this to the roadmap, need to make some changes before we can get people to contribute.

**Comment by katzecz** [comment]
Would be interested to contribute (CS/CZ <=> EN).

**Comment by shichen437** [comment]
Would be interested to contribute (zh-CN <=> EN).

**Comment by AmadeusGraves** [comment]
Same here, when this feature comes out, I will help with spanish.

**Comment by daniel31x13** [comment]
This feature is already out. You just need to create a new `.json` file in the [locales](https://github.com/linkwarden/linkwarden/tree/dev/public/locales) directory with the following format: `public/locales/xx/common.json`, where `xx` is your two-letter language code.

As a reference, please check out the [English](https://github.com/linkwarden/linkwarden/blob/dev/public/locales/en/common.json) version.

Feel free to make a PR if you found any flaws or incomplete translation.

Thanks for the contribution!

**Comment by click0** [comment]
> Maybe use a tool like [Weblate](https://weblate.org/)? It makes contributing translations much easier, especially for new users.

Or use the services of [Crowdin](crowdin.com) or [Transifex.com](transifex.com) ...

**Comment by shichen437** [comment]
Tools like Weblate may be convenient. However, it is not possible to ensure that every contributor has set up their own development environment, which will affect localization in some contexts. Some projects using Weblate will also remind contributors to set up their own development environment before translating, rather than just using Google Translate, but few contributors will pay attention.

**Comment by ochtum** [comment]
Would be interested to contribute (JP <=> EN).

**Comment by click0** [comment]
Would be interested to contribute (UA <=> EN).

**Comment by daniel31x13** [comment]
@codenyte gonna implement it later, thanks for pointing it out!

@click0 Sure go for it! Thanks :)

**Comment by Orca72** [comment]
I made a Russian translation for Linkwarden. How can I send it to you?

**Comment by daniel31x13** [comment]
@Orca72 thanks, you’ll need to create a new pull request from here:

https://github.com/linkwarden/linkwarden/pulls

More info: https://github.com/linkwarden/linkwarden/issues/216#issuecomment-2290240370

**Comment by bennyz327** [comment]
Can I contribute to Traditional Chinese (zh-TW)?

**Comment by leo-jmateo** [comment]
Hello,

I would like to offer my help in translating **Linkwarden** into Catalan (ca-ES) and/or Spanish (es-ES). I see that currently there are no translations available for these languages, and I would be interested in contributing.

If you already have a system in place to manage translations (for example, Crowdin, Transifex, or another tool), I would be happy to follow the established procedure. If not, I can help you start this process.

Thank you for your work, and I look forward to your response!

Best regards,

**Comment by daniel31x13** [comment]
@leo-jmateo sure you can go for the Catalan translation but we already have a Spanish translation merged in dev, feel free to see if you find any issues: 
https://github.com/linkwarden/linkwarden/blob/dev/public/locales/es/common.json

And for the localization software, we'll be choosing Weblate. Will let you know as soon as I set things up, thanks!

**Comment by 7Adrian** [comment]
I translated it into Polish. #897 
@Uncorrupt3318, feel free to review it. 😃 

Also, I found a leading space in the key `stripe_update_note`. It might be a bug, but I didn't verify it.

**Comment by SoCuul** [comment]
From my experience (as an end-user), I've had by **far** the best experience with [CrowdTranslate](https://www.crowdtranslate.net/). It's a newer translation platform made by (and for) smaller developers to provide the least friction to contribute to translations.

It's super streamlined and well-designed (almost apple-esque), costs almost nothing and really makes it much easier for people to contribute. I may sound like I'm getting paid for this comment haha, but I've genuinely had a shockingly good experience while translating for [Replacicon](https://replacicon.app/) which I haven't found any other service provide.

**Comment by CerealKiller97** [comment]
Hi! I would like to make a translation in Serbian (Cyrilic and Latin).

**Comment by daniel31x13** [comment]
Sure, go for it!

**Comment by LordArrin** [comment]
Can help with Russian translation.

**Comment by daniel31x13** [comment]
Hey hey, we now have a [Crowdin page](https://crowdin.com/project/linkwarden) which makes it much easier to translate Linkwarden :)
